### PR TITLE
build/run on machines that only have SSE2 with SIMDE

### LINKED ISSUE #204: use SIMDe to add plain amd64 (SSE2) to the FAT_RUNTIME
This doesn't have to be the default, but it would aid Debian, Fedora, and others who still have to support plain x86-64 systems that only have SSE2

(I tried, but it was a bit beyond me).

Thanks!

### PR DESCRIPTION
**Pull request by isildur-g**
build/run on machines that only have SSE2 with SIMDE

### CODE REVIEW

File: src/dispatcher.c [hunk 1]
```diff
@@ -78,6 +78,12 @@
 #define check_avx512vbmi() (0)
 #endif
 
+#if defined(BUILD_SSE2_SIMDE)
+#define CHECK_SSE_LAST check_ssse2
```
Comment (mr-c) [inline src/dispatcher.c]: Is there a `check_ssse2` function?

Comment (markos) [reply src/dispatcher.c]: @mr-c we will have a base dependency for sse2, there will be no support for anything before SSE2. without SSE2 compilation will fail -for fat runtimes, one can use it for a non-fat runtime with SIMDE_BACKEND. We will fix this in another commit.

File: src/dispatcher.c [hunk 2]
```diff
@@ -78,6 +78,12 @@
 #define check_avx512vbmi() (0)
 #endif
 
+#if defined(VS_SIMDE_BACKEND)
```
Comment (markos) [inline src/dispatcher.c]: no need for these anymore.

File: src/hs.cpp [hunk 3]
```diff
@@ -205,8 +205,8 @@ hs_compile_multi_int(const char *const *expressions, const unsigned *flags,
         *comp_error = generateCompileError("Unsupported architecture", -1);
         return HS_ARCH_ERROR;
     }
-#endif
-#endif
+#endif // arch
+#endif // FAT_RUNTIME
```
Comment (markos) [inline src/hs.cpp]: same, remove comments.

File: src/dispatcher.c [hunk 4]
```diff
@@ -111,7 +117,7 @@
         else if (check_sse42() && check_popcnt()) {                            \
             fat_dispatch_ ## NAME = &JOIN(corei7_, NAME);                      \
         }                                                                      \
-        else if (check_ssse3()) {                                              \
+        else if (CHECK_SSE_LAST()) {                                           \
```
Comment (markos) [inline src/dispatcher.c]: same, just use the function

**Review by markos — APPROVED** [review]
Nice one, we can finally close this.

**Comment by mr-c** [comment]
Thank you @markos !

**Pull request merged**